File: TCTM/tctm.client/src/api/enums.js
/**
 * Enum mirrors for backend values.
 * Keep in sync with TCTM.Server/DataModel/Enums.cs
 */

/** @enum {string} */
export const TournamentFormat = Object.freeze({
  RoundRobin: 'RoundRobin',
  Swiss: 'Swiss',
  SingleElimination: 'SingleElimination',
  DoubleElimination: 'DoubleElimination',
})

/** @enum {string} */
export const TimeControlPreset = Object.freeze({
  Bullet: 'Bullet',
  Blitz: 'Blitz',
  Rapid: 'Rapid',
})

/** @enum {string} */
export const TournamentStatus = Object.freeze({
  Lobby: 'Lobby',
  InProgress: 'InProgress',
  Completed: 'Completed',
})

/** @enum {string} */
export const RoundStatus = Object.freeze({
  Pending: 'Pending',
  InProgress: 'InProgress',
  Completed: 'Completed',
})

/** @enum {string} */
export const MatchResult = Object.freeze({
  WhiteWin: 'WhiteWin',
  BlackWin: 'BlackWin',
  Draw: 'Draw',
})

/** @enum {string} */
export const Bracket = Object.freeze({
  Winners: 'Winners',
  Losers: 'Losers',
})

File: TCTM/TCTM.Server/DataModel/Enums.cs
namespace TCTM.Server.DataModel;

public enum TournamentFormat
{
    RoundRobin,
    Swiss,
    SingleElimination,
    DoubleElimination
}

public enum TimeControlPreset
{
    Bullet,
    Blitz,
    Rapid
}

public enum TournamentStatus
{
    Lobby,
    InProgress,
    Completed
}

public enum RoundStatus
{
    Pending,
    InProgress,
    Completed
}

public enum MatchResult
{
    WhiteWin,
    BlackWin,
    Draw
}

public enum Bracket
{
    Winners,
    Losers
}

public enum LiveGameStatus
{
    NotStarted,
    InProgress,
    Completed,
    Aborted
}
